test/tests/functional/pbs_job_array.py

```diff
@@ -68,3 +69,51 @@ def test_arrayjob_Erecord_startval(self):
         # Verify that the value of 'start' isn't 0
         self.assertNotEqual(start_val, 0,
                             "E record value of 'start' for arrayjob is 0")
+
+    def test_arraysubjob_survive_restart(self):
+        """
+        Test to check if subjobs of an array job survive a pbs_server restart
+        """
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        j = Job(TEST_USER, attrs={
+            ATTR_J: '1-3', 'Resource_List.select': 'ncpus=1'})
+
+        j.set_sleep_time(5)
+
+        j_id = self.server.submit(j)
+
+        # 1. check job array has begun
+        self.server.expect(JOB, {'job_state': 'B'}, j_id)
+
+        # 2. wait till subjob 2 starts running
+        self.server.expect(JOB, {'job_state': 'R'}, create_subjob_id(j_id, 2))
+
+        # 3. Kill the server
+        self.server.stop('-KILL')
+
+        # 4. start the server
+        self.server.start()
+
+        # 5. array job should be B
+        self.server.expect(JOB, {'job_state': 'B'}, j_id, max_attempts=1)
+
+        # 6. subjob 1 should be X
+        self.server.expect(JOB, {'job_state': 'X'}, create_subjob_id(j_id, 1),
+                           max_attempts=1)
+
+        # 7. subjob 2 should be R
+        self.server.expect(JOB, {'job_state': 'R'}, create_subjob_id(j_id, 2),
+                           max_attempts=1)
+
+        # 8. subjob 3 should be Q
+        self.server.expect(JOB, {'job_state': 'Q'}, create_subjob_id(j_id, 3),
+                           max_attempts=1)
+
```

Comment:
Can you please add the following also in the same test case or separate test case.
* do a qdel on one of the subjobs that is running and a one which is just queued and verify them after server restart.
* Also place couple of subjobs in suspend state using qsig and verify them after the restart that they are still in suspend state. Do this for range of subjobs as well.
* Repeat the whole test case with job history enabled.

Reply:
>do a qdel on one of the subjobs that is running and a one which is just queued and verify them after server restart.
Also place couple of subjobs in suspend state using qsig and verify them after the restart that they are still in suspend state. Do this for range of subjobs as well.

The above two proposed test cases are not valid use cases as per the description of the RFE ticket of this PR  [PP-479](https://pbspro.atlassian.net/browse/PP-479)  

>Repeat the whole test case with job history enabled.

Added PTL test case TestJobArray.test_arraysubjob_survive_restart_with_history in rc 26 https://github.com/PBSPro/pbspro/pull/590/commits/c44f23fce932830e60aaafcd206ec0be6fd034ab

Reply:
These test cases will ensure the following.
* If the behavior of suspended/queued jobs after restart is still the same before and after check-in of this RFE.
*  If the behavior is changed then these will will verify the current expected state for the subjobs.

I am fine even if there are existing test cases to verify the above in which case you can just run them or else we need to write the test cases to make sure things are working as expected.

Reply:
As I explained to you, since the Job array was always requeued after the server restart, it wont make sense to have a PTL test case just to compare the behavior of suspended/queued jobs after restart is still same before and after check-in of this RFE.

Reply:
Added the requested PTL tests in rc 30 https://github.com/PBSPro/pbspro/pull/590/commits/50e996cba0451e8eed701dc2cde066effeb8077b